.github/workflows/rust.yml

```diff
@@ -45,11 +45,16 @@ jobs:
 
       - name: Configure cache
         uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # @v3.3.2
+        env:
+          cache-name: cargo-fmt
         with:
           path: |
-            ~/.cargo/registry
-            ~/.cargo/git
+            ${{ runner.os == 'Windows' && format('{0}/.cargo/registry', env.USERPROFILE) || '~/.cargo/registry' }}
+            ${{ runner.os == 'Windows' && format('{0}/.cargo/git', env.USERPROFILE) || '~/.cargo/git' }}
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-${{ env.cache-name }}-
+            ${{ runner.os }}-cargo-
```

Comment:
Generic `${{ runner.os }}-cargo-` is good, but `${{ runner.os }}-cargo-${{ env.cache-name }}-` will not be helpful here since we will never have anything in cache that matches such a key.
Restore keys are only used for matching during retrieval.